# 05 — Valuation: What the Market Is Paying

Notebooks 02 and 03 establish what these businesses earn and how much cash they
generate. This one connects that to price.

Three multiples, each answering a different question:

| Multiple | Question | Direction |
| --- | --- | --- |
| **P/E** = Market Cap / Net Income | What is an investor paying per dollar of profit? | lower = cheaper |
| **FCF Yield** = Free Cash Flow / Market Cap | What cash return does the business throw off on its price? | higher = cheaper |
| **EV/EBITDA** = (Market Cap + Total Debt − Cash) / EBITDA | What is the *whole business* worth, debt included, per dollar of operating earnings? | lower = cheaper |

The third exists because the first two ignore the balance sheet. That distinction
turns out to decide the entire Ford question.

## The currency trap this notebook used to fall into

BYD's ADR trades in **dollars**; BYD reports in **renminbi**. Dividing a dollar
market cap by a renminbi net income produced a P/E of **2.4x** — roughly a seventh
of the truth — and an FCF yield of **36.8%**. Both were quoted as headline findings.

Notebook 01 now translates every BYD figure to USD before it reaches this
notebook, so the multiples below divide dollars by dollars. BYD's real 2024 P/E is
about **17.5x**: still less than a tenth of Tesla's, which is the finding that
survives — but a tenth, not a seventy-eighth.

In [1]:
import sys

sys.path.append("..")

import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

import config

fundamentals = pd.read_pickle("../data/fundamentals.pkl")
YEARS = config.ANALYSIS_YEARS
COLORS = config.COLORS

print(f"Valuing {', '.join(config.COMPANIES)} over {config.FIRST_YEAR}-{config.LAST_YEAR}")

Valuing Tesla, BYD, Ford over 2022-2024


## The multiples

Market capitalisation and all three multiples are computed in notebook 01, using the
tested helpers in `metrics.py` so the formulas cannot drift between notebooks. This
notebook reads them and interprets them.

`price_to_earnings` returns `None` for a loss-making year: a negative P/E is
arithmetically valid and analytically meaningless, which is why Ford's 2022 is blank
rather than a number.

In [2]:
valuation = fundamentals.copy()

multiples = valuation.set_index(["company", "year"])[["pe", "fcf_yield", "ev_ebitda"]]
multiples.columns = ["P/E", "FCF Yield (%)", "EV/EBITDA"]
print("=== VALUATION MULTIPLES (USD) ===")
print(multiples.round(2).to_string())

print("\nYear-end market capitalisation ($B):")
print((valuation.pivot_table(index="year", columns="company",
                             values="market_cap", observed=True) / 1e9).round(1).to_string())
print("\nFord 2022 P/E is blank: the company reported a net loss that year.")

=== VALUATION MULTIPLES (USD) ===
                 P/E  FCF Yield (%)  EV/EBITDA
company year                                  
Tesla   2022   31.04           1.94      21.48
        2023   52.77           0.55      53.03
        2024  189.31           0.27      91.10
BYD     2022   27.96           9.33      10.22
        2023   18.38           8.63       5.95
        2024   17.53           5.12       5.51
Ford    2022     NaN          -0.04      31.68
        2023    9.37          16.40      14.14
        2024    5.99          19.13      12.16

Year-end market capitalisation ($B):
company   Tesla   BYD  Ford
year                       
2022      389.7  69.1  35.4
2023      791.4  78.1  40.7
2024     1342.4  98.2  35.2

Ford 2022 P/E is blank: the company reported a net loss that year.


In [3]:
fig = make_subplots(rows=1, cols=3, subplot_titles=("P/E Ratio", "FCF Yield (%)", "EV/EBITDA"))

for col, field in enumerate(["pe", "fcf_yield", "ev_ebitda"], start=1):
    for name in config.COMPANIES:
        series = valuation.query("company == @name").set_index("year")[field].dropna()
        fig.add_trace(go.Scatter(
            x=series.index, y=series.values, name=name, mode="lines+markers",
            line=dict(color=COLORS[name], width=2), marker=dict(size=8),
            legendgroup=name, showlegend=(col == 1),
        ), row=1, col=col)

fig.update_layout(
    title_text=f"Valuation multiples — Tesla vs BYD vs Ford ({config.FIRST_YEAR}–{config.LAST_YEAR}, USD)",
    template="plotly_white", height=450, width=1200, hovermode="x unified",
)
for col in range(1, 4):
    fig.update_xaxes(tickmode="array", tickvals=YEARS,
                     ticktext=[str(y) for y in YEARS], row=1, col=col)
fig.show()

## Who is actually cheapest?

The heatmap normalises each multiple to a 0–100 scale so that **green always means
better value**, which requires inverting P/E and EV/EBITDA (where lower is cheaper)
but not FCF yield (where higher is).

Read it as a *ranking within these three companies*, not an absolute score: the
scale is min-max, so the cheapest company on any metric scores 100 by construction
and the dearest scores 0, however narrow or wide the real gap.

In [4]:
snapshot = valuation.query("year == @config.LAST_YEAR").set_index("company")
scores = snapshot[["pe", "fcf_yield", "ev_ebitda"]].copy()

# Invert the two where lower is cheaper, so higher = better value everywhere.
scores["pe"] = 1 / scores["pe"]
scores["ev_ebitda"] = 1 / scores["ev_ebitda"]
scores = (scores - scores.min()) / (scores.max() - scores.min()) * 100
scores.columns = ["P/E", "FCF Yield", "EV/EBITDA"]

print(f"{config.LAST_YEAR} value scores (100 = cheapest of the three on that metric):")
print(scores.round(1).to_string())

fig_heat = px.imshow(
    scores, text_auto=".0f", color_continuous_scale="RdYlGn", aspect="auto",
    title=f"Relative value, {config.LAST_YEAR} (100 = cheapest of the three)",
    labels=dict(color="Value score"),
)
fig_heat.update_layout(template="plotly_white", height=300, width=760)
fig_heat.show()

2024 value scores (100 = cheapest of the three on that metric):
           P/E  FCF Yield  EV/EBITDA
company                             
Tesla      0.0        0.0        0.0
BYD       32.0       25.7      100.0
Ford     100.0      100.0       41.8


In [5]:
# The Ford paradox, quantified: cheapest on equity multiples, mid-pack on
# enterprise value, entirely because of Ford Credit's borrowings.
bridge = snapshot[["market_cap", "total_debt", "cash", "enterprise_value", "ebitda"]] / 1e9
bridge["debt / market cap"] = snapshot["total_debt"] / snapshot["market_cap"]
print(f"Market cap to enterprise value, {config.LAST_YEAR} ($B):")
print(bridge.round(2).to_string())

Market cap to enterprise value, 2024 ($B):
         market_cap  total_debt   cash  enterprise_value  ebitda  debt / market cap
company                                                                            
Tesla       1342.39       13.62  16.14           1339.87   14.71               0.01
BYD           98.18        5.54  14.08             89.65   16.26               0.06
Ford          35.24      160.86  22.94            173.16   14.24               4.57


## Key finding

**The market pays about 189x earnings for Tesla, 18x for BYD and 6x for Ford.**

That 31-fold spread between the most and least expensive is the finding, and it
survives the currency correction intact — what changes is BYD's position within it
and, more sharply, Ford's.

**2024, in USD:**

| | Tesla | BYD | Ford |
|---|---:|---:|---:|
| P/E | 189x | 18x | 6x |
| FCF Yield | 0.3% | 5.1% | 19.1% |
| EV/EBITDA | 91x | 5.5x | 12.2x |
| Free cash flow | $3.6B | $5.0B | $6.7B |
| Revenue growth (2022→2024) | +20% | +72% | +17% |

**Tesla is the most expensive company here on every measure**, by a margin that
no longer has much to do with its current financials: it generates the least free
cash flow of the three, grew revenue 0.9% in 2024, and saw margins halve. The 189x
is a claim about FSD, Optimus and energy storage. None of them appear in these
statements — which is not an argument that they won't, only that this data cannot
support the price on its own.

**BYD is cheap relative to what it is doing.** 18x earnings for a business growing
revenue 27% a year, with the best cash conversion of the three and margins still
rising, is the most defensible risk-adjusted position in the set. The discount is a
geopolitical one — tariffs, ADR structure, disclosure quality — not a financial one.

**Ford is where the two lenses disagree, and the disagreement is the point.** It is
the cheapest on P/E (6x) and by far the highest FCF yield (19.1%), yet only
mid-pack on EV/EBITDA (12.2x versus BYD's 5.5x). The reason is $160.9B of debt
against a $35.2B market cap — a 4.6x debt-to-market-cap ratio, almost all of it
Ford Credit's lending book. Equity holders see a cheap stock; the enterprise is not
cheap. Anyone quoting Ford's 19% FCF yield without that context is quoting half a
number.

*This analysis uses three years of annual financial data and excludes forward
guidance, segment detail and market context. It is an analytical exercise, not
investment advice.*

In [6]:
print("Valuation multiples live in data/fundamentals.pkl, written by notebook 01.")

Valuation multiples live in data/fundamentals.pkl, written by notebook 01.
